Dart/src/com/jetbrains/lang/dart/resolve/DartResolver.java

```diff
@@ -22,9 +30,36 @@
 public class DartResolver implements ResolveCache.AbstractResolver<DartReference, List<? extends PsiElement>> {
   public static final DartResolver INSTANCE = new DartResolver();
 
+  public static final boolean USE_SERVER = false;
+
   @Nullable
   @Override
   public List<? extends PsiElement> resolve(@NotNull DartReference reference, boolean incompleteCode) {
+    if (USE_SERVER) {
+      final PsiFile refPsiFile = reference.getContainingFile();
+      //if (!DartAnalysisServerAnnotator.serverReadyForRequest(refPsiFile)) {
+      //  return null;
+      //}
+      final VirtualFile refVirtualFile = refPsiFile.getVirtualFile();
+      if (refVirtualFile != null) {
+        if (isIdentifier(reference)) {
+          Project project = reference.getProject();
+          final int refOffset = reference.getTextOffset();
+          final String refPath = refVirtualFile.getPath();
+          final List<NavigationRegion> regions = DartAnalysisServerService.getInstance().getNavigation(refPath);
+          for (NavigationRegion region : regions) {
+            if (region.containsInclusive(refOffset)) {
+              NavigationTarget target = region.getTargetObjects().get(0);
+              return getResultForNavigationTarget(project, target);
```

Comment:
We can return a list of target PsiElements from this method, so we do not have to ignore other region.getTargetObjects(). BTW what are cases when this list is greater than 1? Duplicate declarations?